cloudmailer/cloudmailer.py

```diff
@@ -255,31 +255,77 @@ def listFile(textfile):
     lfile.close()
     return items
 
+# getServergroupsAndVms
+#
+# The function creates two maps, nodes and servergroups
+# For each node, the map nodes describes the list of server groups which have at least one server hosted in the node.
+# For each server group, the map servergroups describes the list of nodes which host at least one server of the server group.
+#
+# input
+# data: information about the Openstack platform retrieved from API call
+# nodelist: list of nodes we obtained in input
+#
+# output
+# nodes: dict containing [node->list of server groups] mappings
+# servergroups: dict containing [server group->list of nodes] mappings
 def getServergroupsAndVms(data,nodelist):
-
+    # create empty nodes map
     nodes = {}
+    # create empty servergroups map
     servergroups = {}
+    # for each of the nodes we have received in input
     for node in nodelist:
+        # prepare an empty list of server groups for the node
         nodes[node] = []
 
+    # allservers is a collection containing all the servers in the Openstack platform
     allservers = data.all_servers
+    # server_groups is a collection containing all the server groups in the Openstack platform
     server_groups = data.all_server_groups
 
-
     # Generates a servergroup/hypervisor <-> servergroup/instance mapping.
+    #
+    # for each group in the server groups
     for group in server_groups:
+        # prepare an empty list, which will contain the nodes hosting the servers of the server group
         grouphosts = []
+        # for each of the servers in the server group
         for member in group.members:
+            # initialize a flag that tells if we found the server in the list of servers given by the Openstack platform
+            server_found = False
+            # initialize the variable that will contain the node where the server is being hosted
+            host = None
+            # for each of the servers in the Openstack platform
```

Comment:
Possibly superfluous

Reply:
I suppose here you refer to the two comments and not to the actual code